src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -386,8 +392,20 @@ public void StartProfileOptimization(string profile)
             InternalStartProfile(profile, m_pNativeAssemblyLoadContext);
 #endif // FEATURE_MULTICOREJI
         }
-        
+
+        private void OnProcessExit(object sender, EventArgs e)
+        {
+            var unloading = Unloading;
+            if (unloading != null)
+            {
+                unloading(this, EventArgs.Empty);
+            }
+        }
+
         public event Func<AssemblyLoadContext, AssemblyName, Assembly> Resolving;
+        public event EventHandler Unloading;
```

Comment:
I don't think we discussed the event signature. I'll add a couple of people to double-check.